pages/forums/view.js

```diff
@@ -0,0 +1,132 @@
+// Module imports
+import React from 'react'
+
+
+
+
+
+// Component imports
+import { actions } from '../../store'
+import Component from '../../components/Component'
+import ForumThreadCard from '../../components/Forums/ForumThreadCard'
+import Page from '../../components/Page'
+
+
+
+
+
+// Component constants
+const pageTitle = 'Public Forums'
+
+
+
+
+
+class ViewThread extends Component {
+  /***************************************************************************\
+    Public Methods
+  \***************************************************************************/
+
+  async componentDidMount () {
+    const {
+      getForumThread,
+      query,
+    } = this.props
+
+    let {
+      thread,
+    } = this.props
+
+    if (!thread) {
+      const { payload, status } = await getForumThread(query.id)
+
+      if (status === 'success' && payload.data) {
+        thread = payload.data
+      }
+    }
+
+    this.setState({
+      loaded: true,
+      thread,
+    })
+  }
+
+  constructor (props) {
+    super(props)
+
+    const { thread } = props
+    this.state = {
+      loaded: !!thread,
+      thread,
+    }
+  }
+
+  static async getInitialProps ({ query, store }) {
+    await actions.getForumThread(query.id)(store.dispatch)
+    return {}
```

Comment:
In the future, we should make sure we preserve the data passed in to `getInitialProps`. Otherwise we can't reference any of the request params from the props.

Reply:
I'm just copying your group example :D